kyo-prelude/shared/src/main/scala/kyo2/Var.scala

```diff
@@ -4,62 +4,73 @@ import Var.internal.*
 import kyo.Tag
 import kyo2.kernel.*
 
-sealed trait Var[V] extends Effect[Input[V, *], Id]
+sealed trait Var[V] extends Effect[Const[Op[V]], Const[V]]
 
 object Var:
 
+    /** Obtains the current value of the 'Var'. */
     inline def get[V](using inline tag: Tag[Var[V]], inline frame: Frame): V < Var[V] =
         use[V](identity)
 
     final class UseOps[V](dummy: Unit) extends AnyVal:
+        /** Invokes the provided function with the current value of the `Var`. */
         inline def apply[A, S](inline f: V => A < S)(
             using
             inline tag: Tag[Var[V]],
             inline frame: Frame
         ): A < (Var[V] & S) =
-            Effect.suspendMap[V](tag, internal.get[V])(f)
+            Effect.suspendMap[V](tag, Get: Op[V])(f)
     end UseOps
 
     inline def use[V]: UseOps[V] = UseOps(())
 
-    inline def set[V](inline value: V)(using inline tag: Tag[Var[V]], inline frame: Frame): Unit < Var[V] =
-        Effect.suspend[Unit](tag, (() => value): Set[V])
+    /** Sets a new value and returns the previous one. */
+    inline def set[V](inline value: V)(using inline tag: Tag[Var[V]], inline frame: Frame): V < Var[V] =
+        Effect.suspend[Unit](tag, value: Op[V])
 
+    /** Sets a new value and returns `Unit`. */
+    inline def setUnit[V](inline value: V)(using inline tag: Tag[Var[V]], inline frame: Frame): Unit < Var[V] =
```

Comment:
I don't love how it sounds like you are setting the value to `Unit` - perhaps `setDiscard`?

Reply:
Makes sense, I'll start using this nomenclature. Should we keep `pending.unit` or rename it to `pending.discard` for consistency?